### PR TITLE
OP-6382 - Thumbnail Integration Problem

### PR DESCRIPTION
**Pull request by tokejepsen**
## Changelog Description
This ticket alerted to 3 different cases of integration issues;

- [x] Using the Tray Publisher with the same image format (extension) for representation and review representation.
- [x] Clash on publish file path from output definitions in `ExtractOIIOTranscode`.
- [x] Clash on publish file from thumbnail in `ExtractThumbnail`

There might be an issue with this fix, if a studio does not use the `{output}` token in their `render` anatomy template. But thinking if they have customized it, they will be responsible to maintain these edge cases.

## Testing notes:
1. Use Tray Publisher to publish `Image` with two different images of same extension.
image
2. Publish should succeed.

1. Setup `ayon+settings://core/publish/ExtractOIIOTranscode/profiles` with settings below.
Capture
2. Setup single frame render from Maya and publish to Deadline.
3. Publish on Deadline should succeed.

1. Setup `ayon+settings://core/publish/ExtractOIIOTranscode/profiles` with settings below (just adding `review` tag from above settings).
Capture
2. Setup single frame render from Maya and publish to Deadline.
3. Publish on Deadline should succeed.

### CODE REVIEW

**Comment by moonyuet** [comment]
Guess the error above is related to this code: 
https://github.com/ynput/OpenPype/blob/c8e00d4aa4cfc81d967c8d607bd409af9659b068/openpype/plugins/publish/integrate.py#L179-L181

**Comment by LiborBatek** [comment]
Did anotther test step (05) and Im getting error on DL for publish job:

```
2024-01-24 11:33:40:  0: STDOUT: DEBUG:pyblish.IntegrateAsset:Anatomy template name: publish_render
2024-01-24 11:33:40:  0: STDOUT: ERROR:pyblish.plugin:Traceback (most recent call last):
2024-01-24 11:33:40:  0: STDOUT:   File "C:\Users\lbate\AppData\Local\Ynput\AYON\addons\openpype_3.18.5-nightly.2\openpype\plugins\publish\integrate.py", line 183, in process
2024-01-24 11:33:40:  0: STDOUT:   File "C:\Users\lbate\AppData\Local\Ynput\AYON\addons\openpype_3.18.5-nightly.2\openpype\plugins\publish\integrate.py", line 283, in register
2024-01-24 11:33:40:  0: STDOUT:   File "C:\Users\lbate\AppData\Local\Ynput\AYON\addons\openpype_3.18.5-nightly.2\openpype\lib\file_transaction.py", line 98, in add
2024-01-24 11:33:40:  0: STDOUT:     raise DuplicateDestinationError(
2024-01-24 11:33:40:  0: STDOUT: openpype.lib.file_transaction.DuplicateDestinationError: Transfer to destination is already in queue: C:\Users\lbate\AppData\Local\Temp\op_transcoding_cax4llrd\Main.1001.jpg -> C:\projects\Ayon_Tests\maya\props\crate\publish\render\renderGenericMain_beauty\v001\ATS_crate_renderGenericMain_beauty_v001.jpg. It's not allowed to be replaced by a new transfer from C:\Users\lbate\AppData\Local\Temp\pyblish_tmp_9hhlfndf\Main.1001_thumb.jpg
2024-01-24 11:33:40:  0: STDOUT: During handling of the above exception, another exception occurred:
2024-01-24 11:33:40:  0: STDOUT: Traceback (most recent call last):
2024-01-24 11:33:40:  0: STDOUT:   File "C:\Users\lbate\AppData\Local\Ynput\AYON\dependency_packages\ayon_2401161802_windows.zip\dependencies\pyblish\plugin.py", line 527, in __explicit_process
2024-01-24 11:33:40:  0: STDOUT:     runner(*args)
2024-01-24 11:33:40:  0: STDOUT:   File "C:\Users\lbate\AppData\Local\Ynput\AYON\addons\openpype_3.18.5-nightly.2\openpype\plugins\publish\integrate.py", line 188, in process
2024-01-24 11:33:40:  0: STDOUT:   File "six.py", line 718, in reraise
2024-01-24 11:33:40:  0: STDOUT:   File "C:\Users\lbate\AppData\Local\Ynput\AYON\addons\openpype_3.18.5-nightly.2\openpype\plugins\publish\integrate.py", line 183, in process
2024-01-24 11:33:40:  0: STDOUT:   File "C:\Users\lbate\AppData\Local\Ynput\AYON\addons\openpype_3.18.5-nightly.2\openpype\plugins\publish\integrate.py", line 283, in register
2024-01-24 11:33:40:  0: STDOUT:   File "C:\Users\lbate\AppData\Local\Ynput\AYON\addons\openpype_3.18.5-nightly.2\openpype\lib\file_transaction.py", line 98, in add
2024-01-24 11:33:40:  0: STDOUT:     raise DuplicateDestinationError(
2024-01-24 11:33:40:  0: STDOUT: openpype.pipeline.publish.publish_plugins.KnownPublishError: Transfer to destination is already in queue: C:\Users\lbate\AppData\Local\Temp\op_transcoding_cax4llrd\Main.1001.jpg -> C:\projects\Ayon_Tests\maya\props\crate\publish\render\renderGenericMain_beauty\v001\ATS_crate_renderGenericMain_beauty_v001.jpg. It's not allowed to be replaced by a new transfer from C:\Users\lbate\AppData\Local\Temp\pyblish_tmp_9hhlfndf\Main.1001_thumb.jpg
2024-01-24 11:33:40:  0: STDOUT: Traceback (most recent call last):
2024-01-24 11:33:40:  0: STDOUT:   File "<string>", line 183, in process
2024-01-24 11:33:40:  0: STDOUT:   File "<string>", line 283, in register
2024-01-24 11:33:40:  0: STDOUT:   File "C:\Users\lbate\AppData\Local\Ynput\AYON\addons\openpype_3.18.5-nightly.2\openpype\lib\file_transaction.py", line 98, in add
2024-01-24 11:33:40:  0: STDOUT:     raise DuplicateDestinationError(
2024-01-24 11:33:40:  0: STDOUT: openpype.lib.file_transaction.DuplicateDestinationError: Transfer to destination is already in queue: C:\Users\lbate\AppData\Local\Temp\op_transcoding_cax4llrd\Main.1001.jpg -> C:\projects\Ayon_Tests\maya\props\crate\publish\render\renderGenericMain_beauty\v001\ATS_crate_renderGenericMain_beauty_v001.jpg. It's not allowed to be replaced by a new transfer from C:\Users\lbate\AppData\Local\Temp\pyblish_tmp_9hhlfndf\Main.1001_thumb.jpg
2024-01-24 11:33:40:  0: STDOUT: During handling of the above exception, another exception occurred:
2024-01-24 11:33:40:  0: STDOUT: Traceback (most recent call last):
2024-01-24 11:33:40:  0: STDOUT:   File "C:\Users\lbate\AppData\Local\Ynput\AYON\dependency_packages\ayon_2401161802_windows.zip\dependencies\pyblish\plugin.py", line 527, in __explicit_process
2024-01-24 11:33:40:  0: STDOUT:     runner(*args)
2024-01-24 11:33:40:  0: STDOUT:   File "<string>", line 188, in process
2024-01-24 11:33:40:  0: STDOUT:   File "six.py", line 718, in reraise
2024-01-24 11:33:40:  0: STDOUT:   File "<string>", line 183, in process
2024-01-24 11:33:40:  0: STDOUT:   File "<string>", line 283, in register
2024-01-24 11:33:40:  0: STDOUT:   File "C:\Users\lbate\AppData\Local\Ynput\AYON\addons\openpype_3.18.5-nightly.2\openpype\lib\file_transaction.py", line 98, in add
2024-01-24 11:33:40:  0: STDOUT:     raise DuplicateDestinationError(
2024-01-24 11:33:40:  0: STDOUT: openpype.pipeline.publish.publish_plugins.KnownPublishError: Transfer to destination is already in queue: C:\Users\lbate\AppData\Local\Temp\op_transcoding_cax4llrd\Main.1001.jpg -> C:\projects\Ayon_Tests\maya\props\crate\publish\render\renderGenericMain_beauty\v001\ATS_crate_renderGenericMain_beauty_v001.jpg. It's not allowed to be replaced by a new transfer from C:\Users\lbate\AppData\Local\Temp\pyblish_tmp_9hhlfndf\Main.1001_thumb.jpg
2024-01-24 11:33:40:  0: STDOUT: !!! ERR: 2024-01-24 11:33:40,083 >>> { CLI-publish }: [ Failed IntegrateAsset: Transfer to destination is already in queue: C:\Users\lbate\AppData\Local\Temp\op_transcoding_cax4llrd\Main.1001.jpg -> C:\projects\Ayon_Tests\maya\props\crate\publish\render\renderGenericMain_beauty\v001\ATS_crate_renderGenericMain_beauty_v001.jpg. It's not allowed to be replaced by a new transfer from C:\Users\lbate\AppData\Local\Temp\pyblish_tmp_9hhlfndf\Main.1001_thumb.jpg -- ('C:\\Users\\lbate\\AppData\\Local\\Ynput\\AYON\\addons\\openpype_3.18.5-nightly.2\\openpype\\plugins\\publish\\integrate.py', 98, 'add', 'raise DuplicateDestinationError(') ] 
2024-01-24 11:33:40:  0: STDOUT: --- your system is set to use custom CA certificate bundle.
```

**Comment by LiborBatek** [comment]
In the last testing step with ```review``` tag added, it ends pretty much the same with failure of the DL publish job with the exact same error msg. so check that above too.

**Comment by tokejepsen** [comment]
> Did test the first case using Tray Publisher

@LiborBatek I've replicated here and it works. I can see what you are using `Production` bundle, could you let me know how you build and are setup, so I can replicate here?

**Comment by LiborBatek** [comment]
> > Did test the first case using Tray Publisher
> 
> @LiborBatek I've replicated here and it works. I can see what you are using `Production` bundle, could you let me know how you build and are setup, so I can replicate here?

You can build from within your OP Repo by 

```OpenPype\server_addon> python .\create_ayon_addons.py```

then just locate the folder ```\server_addon\packages\``` and upload those needed packages on Ayon server and put those into new Addon Bundle

**Review by antirotor — APPROVED** [review]
Tested all combinations - same image, two different images, jpgs, pngs, representation and little thumbnail and all works! Thanks.

> [!NOTE]
> Tested using dev-mode

**Review by LiborBatek — APPROVED** [review]

**Comment by LiborBatek** [comment]
@tokejepsen ok I re did the test again and double checked my addons being correctly set and in ```production``` and finally also works for me. sorry for confusion (probably didnt deleted local ayon data folder content before!)

Went smoothly in Tray Publisher and with thumbnails integrated on Ayon too
image

and on Ayon:
image

File: openpype/plugins/publish/extract_color_transcode.py [hunk 1]
```diff
@@ -189,6 +189,13 @@ def process(self, instance):
                 if len(new_repre["files"]) == 1:
                     new_repre["files"] = new_repre["files"][0]
 
+                # If the source representation has "review" tag, but its not
+                # part of the output defintion tags, then both the
+                # representations will be transcoded in ExtractReview and
+                # their outputs will clash in integration.
+                if not added_review and "review" in repre.get("tags", []):
```
Comment (iLLiCiTiT) [inline openpype/plugins/publish/extract_color_transcode.py:196]:
```suggestion
                if "review" in repre.get("tags", []):
```

**Pull request merged**